sources/licenses/Licenses.yml

```diff
@@ -45,12 +48,36 @@ swap:
   CC[ -]?BY[ -]?SA[ -]?4(?:\.0)?: CC BY-SA 4.0
   CC[ -]?[O0]: CC0
   CC[ -]?0[ -]?1(?:\.0)?: CC0 1.0
-  # GNU Licences
+  # GNU licences
   AGPL: AGPL
   AGPL[ -]?v?3(?:\.0)?: AGPLv3
   GPL(?!\.org): GPL
   GPL[ -]?v?2(?:\.0)?: GPLv2
```

Comment:
@ccoVeille can you please help me here? The rule is quite useful but I haven't been able to modify it to ignore the SPDX identifiers: ``GPL-2.0-only`` or ``GPL-2.0-or-later``.  The rule as is, modifies the ``GPL-2`` part of those identifiers which is undesirable.

The easiest solution would be:
```suggestion
  GPL[ ]?v?2(?:\.0)?: GPLv2
```
but then, we'd lose the capability to correct one of the most popular mistakes, ``GPL-2``. I tried to utilize the fact that ``GPL-2*-*`` should be avoided since that's almost always going to be the SPDX identifier. However, I seem to be doing something wrong and haven't been able to solve it yet.

(Naturally, the changes here will also be replicated for the other GNU licenses)

Reply:
it can be split like this, it would help to maintain the regexp readability I think

```suggestion
  GPL ?v?2(?:\.0)?: GPLv2
  GPL-?v?2(?:\.0)?(?!\-only|\-or\-later): GPLv2
```

Reply:
![image](https://github.com/user-attachments/assets/fee841e0-392e-44aa-a7dd-ae9b234c3d2c)

Reply:
another solution is to split again

   GPL ?v?2(?:\.0)?: GPLv2
   GPL\-?v?2\.0: GPLv2
   GPL\-?v?2(?!\-only|\-or\-later): GPLv2


![image](https://github.com/user-attachments/assets/1cbc9d9d-9c14-40ef-8cce-4b9936e82f8c)

Reply:
```suggestion
  GPL ?v?2(?:\.0)?: GPLv2                   # GPL followed by a space
  GPL\-?v?2\.0: GPLv2                       # GPL followed by -2.0 or -v2.0
  GPL\-?v?2(?!\-only|\-or\-later): GPLv2    # GPL followed by -2 or -v2 but not a known SPDX identifer
```

Reply:
The SPDX identifier is a bit different, you can use this for tests:
```
## Fix

GPL2
GPL 2
GPL-2

## Avoid

GPL-2.0-only
GPL-2.0-or-later
```

Reply:
It seems like adding ``.*(?!only|or-later)`` to the existing GPL corrections suffices.

Reply:
Of course, it does. But you will allow all these false-negatives

GPLv2-only
GPL-v2-only
GPL 2-only
GPL-2-only
GPL v2-only
GPL2-only
GPLv2.0-only
GPL-v2.0-only
GPL 2.0-only
GPL2.0-only
GPL-v2.0-only
GPLv2-or-later
GPL-v2-or-later
GPL v2-or-later
GPL 2-or-later
GPL2-or-later
GPL-v2-or-later
GPLv2.0-or-later
GPL 2.0-or-later
GPL2.0-or-later
GPL 2.0-or-later

So more than allowing only the SPXD identifiers you wanted to allow:

GPL-2.0-only
GPL-2.0-or-later

Reply:
This is the only regexp I would recommend using

https://github.com/jargonLint/jargonLint/pull/58#discussion_r1677885763

Reply:
Let's discuss this in your comment. Our respective comments are too scattered as is.